scanpipe/pipes/purldb.py

```diff
@@ -300,28 +301,43 @@ def get_unique_resolved_purls(project):
 def get_unique_unresolved_purls(project):
     """Return PURLs from project's unresolved DiscoveredDependencies."""
     packages_unresolved = project.discovereddependencies.filter(
-        is_pinned=False
+        is_pinned=False,
     ).exclude(extracted_requirement="*")
 
-    distinct_unresolved_results = packages_unresolved.values(
+    distinct_unpinned_results = packages_unresolved.values(
         "type", "namespace", "name", "extracted_requirement"
     )
 
-    distinct_unresolved = {tuple(item.values()) for item in distinct_unresolved_results}
+    distinct_unpinned = {tuple(item.values()) for item in distinct_unpinned_results}
 
     packages = set()
-    for item in distinct_unresolved:
+    for item in distinct_unpinned:
         pkg_type, namespace, name, extracted_requirement = item
         if range_class := RANGE_CLASS_BY_SCHEMES.get(pkg_type):
+            purl = PackageURL(type=pkg_type, namespace=namespace, name=name)
+
             try:
                 vers = range_class.from_native(extracted_requirement)
-            except InvalidVersionRange:
+            except (InvalidVersionRange, InvalidVersion) as exception:
+                if exception is InvalidVersionRange:
+                    description = "Invalid VersionRange"
+                else:
+                    description = "Invalid exctracted requirement"
```

Comment:
Can't we use the message from the exception instead of this?

Reply:
I thought it would be a better idea to write a short description since we are already passing the message from the exception to the `exception` argument by:
```
project.add_error(
    description=description,
    model="get_unique_unresolved_purls",
    details=details,
    exception=exception,
)
```
But do you mean use a part/slightly modified version of the text from the exception?
Like from the exception: `'npm:wrap-ansi@^7.0.0' is not a valid <class 'univers.versions.SemverVersion'>` would using something the following like be better:
* `Not a valid version in univers`
* `Extracted requirement is not a valid version`

Reply:
> Not a valid version in univers
> Extracted requirement is not a valid version

That sounds good. You can go ahead with those.

Reply:
Done, thanks!